pymc_marketing/mlflow.py

```diff
@@ -531,29 +887,33 @@ def autolog(
     def patch_sample(sample):
         @wraps(sample)
         def new_sample(*args, **kwargs):
-            idata = sample(*args, **kwargs)
-            mlflow.log_param("pymc_marketing_version", __version__)
-            mlflow.log_param("pymc_version", pm.__version__)
-            mlflow.log_param("nuts_sampler", kwargs.get("nuts_sampler", "pymc"))
-
-            # Align with the default values in pymc.sample
-            tune = kwargs.get("tune", 1000)
-
-            if log_sampler_info:
-                log_sample_diagnostics(idata, tune=tune)
-                log_arviz_summary(
-                    idata,
-                    "summary.html",
-                    var_names=summary_var_names,
-                    **arviz_summary_kwargs,
-                )
-
             model = pm.modelcontext(kwargs.get("model"))
-            if log_model_info:
-                log_model_derived_info(model)
+            with model:
+                idata = sample(*args, **kwargs)
+                mlflow.log_param("pymc_marketing_version", __version__)
+                mlflow.log_param("pymc_version", pm.__version__)
+                mlflow.log_param("nuts_sampler", kwargs.get("nuts_sampler", "pymc"))
```

Comment:
Sorry not sure I understand this one - do you mean something like this:

```
def log_sample_diagnostics(
    idata: az.InferenceData,
    tune: int | None = None,
) -> None:
    # ... (existing code) ...

    if inference_library := posterior.attrs.get("inference_library"):
        mlflow.log_param("inference_library", inference_library)
        mlflow.log_param(
            "inference_library_version",
            posterior.attrs["inference_library_version"],
        )
    mlflow.log_param("arviz_version", posterior.attrs["arviz_version"])
    # move this here
    mlflow.log_param("pymc_marketing_version", __version__)

# ... (rest of the file)

def patch_sample(sample):
    @wraps(sample)
    def new_sample(*args, **kwargs):
        model = pm.modelcontext(kwargs.get("model"))
        with model:
            idata = sample(*args, **kwargs)
            
            # Remove these lines as they're now handled in log_sample_diagnostics
            # mlflow.log_param("pymc_marketing_version", __version__)
            # mlflow.log_param("pymc_version", pm.__version__)
            
            mlflow.log_param("nuts_sampler", kwargs.get("nuts_sampler", "pymc"))

            # ... (rest of the function)

    return new_sample
```

Reply:
Those are the areas I had in mind. Are some of them referring to the same values and be duplicated? I just want to consolidate if we can

Reply:
Hmmm there seems to be something I don't understand / have context for in the tests - seems like we need to log the pymc version differently depending on what sampler is used? Would have to change the tests quite a bit if we wanted to only log one of `pymc_version` and `inference_library_version`.

I did add the sampler logging to the MMM `fit` patch though, this was missing haha

Reply:
The inference library will not always be pymc. For instance it can be nutpie and the `inference_library_version` would be the version of nutpie instead of pymc